secboot/secboot_sb.go

```diff
@@ -124,14 +125,17 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 
 	res.PartDevice = partDevice
 
-	keyData, _, err := readKeyFile(sealedEncryptionKeyFile)
-	if err != nil {
-		return res, err
-	}
-
 	var keys []*sb.KeyData
-	if keyData != nil {
-		keys = append(keys, keyData)
+
+	keyData, _, err := readKeyFile(sealedEncryptionKeyFile)
+	if err == nil {
+		if keyData != nil {
+			keys = append(keys, keyData)
+		}
+	} else {
+		if !os.IsNotExist(err) {
+			logger.Noticef("WARNING: there was an error loading key %s: %v", sealedEncryptionKeyFile, err)
```

Comment:
won't we always get this for a while?  it's the plan later to detect earlier that we won't have key files?

Reply:
No, we will not have this because we will not have errors. Then when we move to tokens, we will have ErrNotExist because the files will not be created.

But I want to be safe here, because it could be that we have some broken factory reset that leaves a key file behind. So this is why I want to continue and try to unlock. But still show the warning so we can find out what was the real error if the unlocking did not work.